nncf/nncf_network.py

```diff
@@ -74,12 +74,14 @@ class InsertionInfo:
     def __init__(self, op_exec_context: OperationExecutionContext,
                  is_input=False,
                  is_output=False,
-                 shape_to_operate_on=None):
+                 shape_to_operate_on=None,
+                 adjacent_quantizers_locations: 'AdjacentQuantizersLocations' = None):
```

Comment:
I am against this kind of extension. The new field is quantizer algo-specific, while the rest of the insertion info is more or less compression algorithm-agnostic. Either subclass the `InsertionInfo` struct, or use another means of passing the info between the propagation stage and the quantizer setup stage.

The naming should also be improved. It is not immediately obvious what kind of "adjacency" is meant in the name.

Reply:
Subclassed and renamed